llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -2504,5 +2504,10 @@ RISCVTTIImpl::enableMemCmpExpansion(bool OptSize, bool IsZeroCmp) const {
     Options.LoadSizes = {8, 4, 2, 1};
   else
     Options.LoadSizes = {4, 2, 1};
+  if (IsZeroCmp && ST->hasVInstructions()) {
+    unsigned RealMinVLen = ST->getRealMinVLen() / 8;
+    for (int LMUL = 1; LMUL <= 8; LMUL *= 2)
```

Comment:
Why do we want to limit this to LMUL>=1? Shouldn't we be able to do this for small vectors even when VLEN is large?